devnet-sdk/controller/kt/kt.go

```diff
@@ -0,0 +1,51 @@
+package kt
+
+import (
+	"context"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/controller/surface"
+	"github.com/ethereum-optimism/optimism/kurtosis-devnet/pkg/kurtosis/api/run"
+)
+
+type KurtosisControllerSurface struct {
+	runner *run.KurtosisRunner
+}
+
+func NewKurtosisControllerSurface(enclave string) (*KurtosisControllerSurface, error) {
+	runner, err := run.NewKurtosisRunner(run.WithKurtosisRunnerEnclave(enclave))
+	if err != nil {
+		return nil, err
+	}
+	return &KurtosisControllerSurface{
+		runner: runner,
+	}, nil
+}
+
+func (s *KurtosisControllerSurface) StartService(ctx context.Context, serviceName string) error {
+	script := `
+def run(plan):
+	plan.start_service(name="` + serviceName + `")
+`
+	// start_service is not idempotent, and doesn't return a typed error,
+	// so we need to check the error message
+	if err := s.runner.RunScript(ctx, script); err != nil {
+		msg := err.Error()
+		if strings.Contains(msg, "is already in use by container") {
```

Comment:
Ah this is just like parsing `revert` errors from smart contracts. Let's at least lowerCase `msg` to stay on the safe side